### PR TITLE
V1.1 - GOLD-883: iOS Scroll Fix V2

### PR DESCRIPTION
**Pull request by jansepar**
Merge checklist:
- [x] Removes the old iOS 8.0 patch
- [x] Adds the new iOS 8.0 patch
- [x] Uses the correct iOS 8.0 regex (that works with the simulator)
- [x] Calls the new patch
- [x] Has a test for the new patch
- [x] Has a test for the iOS 8.0 form sibblings bug

### CODE REVIEW

File: api/main.js [hunk 1]
```diff
@@ -92,17 +92,31 @@ $.extend(Mobify.transform, {
...
+
+                // In Webkit, `document.write` immediately executes inline scripts 
+                // not preceded by an external resource.
+                document.write(markup);
+                document.close();
+            });
+        };
+
+        if (Mobify.isIOS8_0()){
+            // See `utils.js` for more information about this fix.
```
Comment (scalvert) [inline api/main.js]: I believe the file is `util.js`

File: api/util.js [hunk 2]
```diff
@@ -40,54 +40,22 @@ Mobify.isIOS8_0 = function() {
...
-
-    return htmlString.replace(BODY_REGEX, openingBodyTag + script);
+Mobify.ios8_0ScrollFix = function(doc) {
+    var meta = document.createElement('meta');
+    meta.setAttribute('name', 'viewport');
+    meta.setAttribute('content', 'width=device-width');
+    // Using `getElementsByTagName` here because grabbing head using
+    // document.head will throw exceptions in some older browsers
+    // (iOS 4.3).
+    doc.getElementsByTagName('head')[0].appendChild(meta);
```
Comment (johnboxall) [inline api/util.js]: @jansepar Probably pretty rare, but you might want to guard against the case when there is no `<head>` element.

Comment (jansepar) [inline api/util.js]: Pretty sure that is impossible. Even if you didn't have a head in your markup, the browser will create it for you.

But yeah, doesn't hurt to guard against it just incase :)

File: api/main.js [hunk 3]
```diff
@@ -92,17 +92,31 @@ $.extend(Mobify.transform, {
...
+            Mobify.ios8_0ScrollFix(document);
+            window.requestAnimationFrame(function(){
+                window.requestAnimationFrame(function(){
```
Comment (jansepar) [inline api/main.js]: FYI, we'll probably refactor these calls to `requestAnimationFrame` into the `ios8_0ScrollFix` itself. Incase anyone was going to comment on that :)

File: tests/runner.sh [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
 PID=$!
+
 sleep 1
 phantomjs tests/phantom.js | grep '<*>' | tee report.xml
-kill $PID
\ No newline at end of file
+ret_code=$?
+
+kill $PID
+exit $ret_code
```
Comment (jansepar) [inline tests/runner.sh:10]: What's the dealio with changing this?

**Comment by jansepar** [comment]
:+1: looks good. Let's merged this once we get approval from QA. We're good to spread this out to 1.0, 1.3, and 2.0!

**Comment by MikeKlemarewski** [comment]
Also tested this scroll patch on Android, iOS 7 and iOS 8. Seems to work!

**Pull request merged**